docs/source/rest-api/openapi.yml

```diff
@@ -0,0 +1,166 @@
+            The filename to upload. Must be a valid IMAP filename that follows the conventions described
+            `here <https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions>`_,
+            *e.g.* ``imap_codice_l1a_lo_20260101_20260102_v01-01.cdf``
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful upload
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+                  example: 's3://bucket-name/filepath/filename.pkts'
+        '400':
+          description: Invalid or missing filename
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  example: 'A pre-signed URL could not be generated. Please ensure that the file name matches mission file naming conventions.'
+
+
+  '/download/{filename_uri}':
+    get:
+      tags:
+        - Download
+      summary: Download files from the SDC
+      operationId: download
+      parameters:
+        - in: path
+          name: s3_uri
+          description: The filename to download from the S3 bucket (i.e. ``s3://{bucket_name}/{filepath}/{filename}``)
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful download
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+                  example: 'Success'
+        '302':
+          description: Redirect needed
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+                  example: '{"download_url": "{pre_signed_url}"}'
+        '400':
+          description: Unsuccessful download
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+                  format: uri
+                  example: 'Not a valid S3 URI.  Example input: s3://bucket/path/file.ext'
+
+  '/query':
+    get:
+      tags:
+        - Query
+      summary: Query for file metadata
+      operationId: query
+      parameters:
+        - in: query
+          name: mission
+          description: The mission of interest (currently only ``imap`` is supported)
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: level
+          description: |
+            The level of data product (e.g. ``l1a``), or ``*`` for all levels. Supported levels are listed
```

Comment:
Do we need "*"? I would say we should just leave the parameter off the request to make it a wildcard instead.